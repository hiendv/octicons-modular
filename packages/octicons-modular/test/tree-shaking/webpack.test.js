/**
 * @jest-environment node
 */

import path from 'path'
import webpack from 'webpack'
import MemoryFS from 'memory-fs'

describe('Webpack tree-shaking', () => {
  test('works', done => {
    let fs = new MemoryFS()
    let compiler = webpack({
      entry: path.resolve(__dirname, 'main.js'),
      output: {
        filename: 'main.js',
        path: '/'
      },
      optimization: {
        sideEffects: true
      },
      mode: 'production'
    })

    compiler.outputFileSystem = fs

    compiler.run((err, stats) => {
      expect(err).toBeFalsy()

      let code = fs.readFileSync('/main.js').toString('utf8')
      expect(code).toMatchSnapshot()
      done()
    })
  })
})
