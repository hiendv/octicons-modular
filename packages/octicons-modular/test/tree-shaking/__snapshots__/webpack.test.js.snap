// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpack tree-shaking works 1`] = `"!function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&\\"object\\"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,\\"default\\",{enumerable:!0,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=0)}([function(e,t,n){\\"use strict\\";n.r(t);var r=function(e,t){const{width:n,height:r,path:o}=t,i=(t,n)=>{n.label?t[\\"aria-label\\"]=n.label:t[\\"aria-hidden\\"]=!0,n.class?t.class=\`octicon octicon-\${e} \${n.class}\`:t.class=\`octicon octicon-\${e}\`;let r=0===n.scale?0:parseFloat(n.scale)||1,o=r*parseInt(t.width),i=r*parseInt(t.height);return t.width=Number(o.toFixed(2)),t.height=Number(i.toFixed(2)),t},l=e=>Object.keys(e).map(t=>\`\${t}=\\"\${e[t]}\\"\`).join(\\" \\").trim();return{name:e,data:t,svg(e,t=document){let a=t.createElement(\\"div\\");return a.innerHTML=\`<svg \${(e=>{let t=Object.assign({},{scale:1,label:null,class:null},e),o=i({version:\\"1.1\\",width:n,height:r,viewBox:\`0 0 \${n} \${r}\`},t);return l(o)})(e)}>\${o}</svg>\`,a.firstChild}}}(\\"alert\\",{keywords:[\\"warning\\",\\"triangle\\",\\"exclamation\\",\\"point\\"],path:'<path fill-rule=\\"evenodd\\" d=\\"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z\\"></path>',width:16,height:16});console.log(r.svg())}]);"`;
